roofit/roofitcore/test/TestStatistics/testLikelihoodGradientJob.cxx

```diff
@@ -589,3 +588,413 @@ TEST_P(LikelihoodGradientJobTest, Gaussian1DAlsoWithLikelihoodJob)
+      // error since Poisson(N|0) is undefined
+      th_sig->SetBinContent(1, 0);
+      th_bkg->SetBinContent(1, 0);
+   }
+
+   RooWorkspace w("w");
+   auto x = w.factory("x[0,10]");
+   w.factory("index[A,B]");
+
+   dynamic_cast<RooRealVar *>(x)->setBins(10);
+
+   // we have to build a simultaneous binned likelihood to trigger the "binnedL" evaluation path
+
+   RooDataHist h_sigA("h_sigA", "h_sigA", *w.var("x"), th_sig.get());
+   RooDataHist h_sigB("h_sigB", "h_sigB", *w.var("x"), th_sig.get());
+   RooDataHist h_bkg("h_bkg", "h_bkg", *w.var("x"), th_bkg.get());
+
+   w.import(h_sigA);
+   w.import(h_sigB);
+   w.import(h_bkg);
+   w.factory("HistPdf::sigA(x,h_sigA)");
+   w.factory("HistPdf::sigB(x,h_sigB)");
+   w.factory("HistPdf::bkg(x,h_bkg)");
+
+   w.factory("ASUM::model_A(mu_sig[1,-1,10]*sigA,mu_bkg_A[1,-1,10]*bkg)");
+   w.factory("ASUM::model_B(mu_sig*sigB,mu_bkg_B[1,-1,10]*bkg)");
+
+   w.pdf("model_A")->setAttribute("BinnedLikelihood");
+   w.pdf("model_B")->setAttribute("BinnedLikelihood");
+
+   // Construct simulatenous pdf
+   w.factory("SIMUL::model(index[A,B],A=model_A,B=model_B)");
+
+   // Construct dataset
+   std::map<std::string, TH1 *> th_data_2D;
+   th_data_2D["A"] = th_data.get();
+   th_data_2D["B"] = th_data.get();
+   RooDataHist h_data("h_data", "h_data", *w.var("x"), *w.cat("index"), th_data_2D);
+
+   // store initial parameters for reuse in second fit
+   std::unique_ptr<RooArgSet> values(w.pdf("model")->getParameters(h_data));
+   RooArgSet savedValues;
+   values->snapshot(savedValues);
+
+   // legacy RooFit fit
+   std::unique_ptr<RooAbsReal> nll(w.pdf("model")->createNLL(h_data, RooFit::EvalBackend::Legacy()));
+   RooMinimizer m0{*nll};
+
+   double nll0BeforeFit = nll->getVal();
+
+   m0.setStrategy(0);
+   m0.setPrintLevel(-1);
+   m0.setVerbose(false);
+   //   m0.setPrintEvalErrors(200);
+
+   m0.minimize("Minuit2", "migrad");
+
+   std::unique_ptr<RooFitResult> m0result{m0.save()};
+   double minNll0 = m0result->minNll();
+   double mu_sig0 = w.var("mu_sig")->getVal();
+   double mu_bkg_A0 = w.var("mu_bkg_A")->getVal();
+   double mu_bkg_B0 = w.var("mu_bkg_B")->getVal();
+
+   values->assign(savedValues);
+
+   std::unique_ptr<RooAbsReal> likelihoodAbsReal{w.pdf("model")->createNLL(h_data, RooFit::ModularL(true))};
+
+   RooMinimizer::Config cfg;
+   cfg.parallelize = NWorkers;
+   cfg.enableParallelDescent = parallelLikelihood;
+   //   cfg.printEvalErrors = 200;
+   RooMinimizer m1(*likelihoodAbsReal, cfg);
+
+   double nll1BeforeFit = likelihoodAbsReal->getVal();
+
+   m1.setStrategy(0);
+   m1.setPrintLevel(-1);
+
+   m1.setVerbose(false);
+
+   m1.minimize("Minuit2", "migrad");
+
+   std::unique_ptr<RooFitResult> m1result{m1.save()};
+   double minNll1 = m1result->minNll();
+   double mu_sig1 = w.var("mu_sig")->getVal();
+   double mu_bkg_A1 = w.var("mu_bkg_A")->getVal();
+   double mu_bkg_B1 = w.var("mu_bkg_B")->getVal();
+
+   EXPECT_EQ(minNll0, minNll1);
```

Comment:
```suggestion
   EXPECT_DOUBLE_EQ(minNll0, minNll1);
```

This comparison fails on `mac-beta`:
```txt
2024-06-26T15:15:19.6793790Z /Users/sftnight/ROOT-CI/src/roofit/roofitcore/test/TestStatistics/testLikelihoodGradientJob.cxx:899: Failure
2024-06-26T15:15:19.6794320Z Expected equality of these values:
2024-06-26T15:15:19.6794520Z   minNll0
2024-06-26T15:15:19.6794660Z     Which is: 110.212
2024-06-26T15:15:19.6794820Z   minNll1
2024-06-26T15:15:19.6794960Z     Which is: 110.212
```
Maybe changing to `EXPECT_DOUBLE_EQ` would fix it?

Reply:
Yes, thanks, I'm on it. I think the fix is actually to (again) switch to component-based splitting. (We should really change the default from event to component based splitting, but that's for a next PR.) I will push it later.